### ISSUE TITLE
Use Sealed Trait As Base Class?

### ISSUE THREAD

**Issue by dalegaspi** [eed3si9n/scalaxb#501]
I'm trying to use jsoniter-scala with the case classes generated using this awesome tool.  the hindrance that i see is that for cases like [this](http://scalaxb.org/running-scalaxb) a `trait` is generated, but the problem is i cannot use jsoniter-scala macros to generate decoders because the [base `trait` is not sealed](https://github.com/plokhotnyuk/jsoniter-scala/issues/247).  can we have perhaps an option to make the generated base class `sealed` for this purpose?

**Comment by plokhotnyuk** [comment]
@dalegaspi You can try to add the `sealed` keyword before `trait` [here](https://github.com/mrdziuban/scalaxb/blob/e3e239766025076f43ec9fa58ef7ce7fa927dd0b/cli/src/main/scala/scalaxb/compiler/xsd/GenSource.scala#L149), build the tool and publish locally to check if it works for your case.

**Comment by dalegaspi** [comment]
@plokhotnyuk yes i saw that...i was going to make the change but then i realized why this was not the default behavior: in a case where the parent class is in a different .xsd file as the child class (which applies in my case), this poses a problem since a sealed trait and its implementation needs to be in the same .scala file...there's a caveat of making sure to create an "intermediate class" that can be inherited in another xsd file.  good thing there is code in there to indicate if the trait is not a root and ensure that it doesn't apply `sealed` keyword.

right now i'm doing it the lazy way and do a search and replace (to add `sealed`) in the generated .scala source files... if i don't run to other issues then i will pursue further.

**Comment by dalegaspi** [comment]
update: looks like manually adding the appropriate `sealed` to the root traits did the trick! 😊😊😊 ...the generated classes now work with jsoniter-scala macros!!  

now i'm just going to make changes to the plugin and make a PR to have an option to make the root traits sealed.

cc: @plokhotnyuk
